actionpack/lib/action_controller/railtie.rb

```diff
@@ -67,5 +67,12 @@ class Railtie < Rails::Railtie #:nodoc:
         config.compile_methods! if config.respond_to?(:compile_methods!)
       end
     end
+
+    initializer "setup session store" do |app|
+      ActiveSupport.on_load(:action_controller) do
+        app_name = app.class.name ? app.railtie_name.sub(/_application$/, '') : ''
```

Comment:
Could make good use of `chomp`: `app.railtie_name.chomp('_application')`.

Also when won't the app class have a `name`? Is it likely that `Rails::Application` subclasses are an anonymous class?